src/libs/kata-sys-util/src/mount.rs

```diff
@@ -0,0 +1,1029 @@
+            ) {
+                Ok(_) => tx.store(true, Ordering::Release),
+                Err(e) => error!(sl!(), "failed to mount in chdir {}: {}", chdir.display(), e),
+            }
+            match unistd::chdir(&cwd) {
+                Ok(_) => info!(sl!(), "chdir from {} to {}", chdir.display(), cwd.display()),
+                Err(e) => {
+                    error!(
+                        sl!(),
+                        "failed to chdir from {} to {} error {:?}",
+                        chdir.display(),
+                        cwd.display(),
+                        e
+                    );
+                }
+            }
+        })?;
+    child.join().map_err(|e| Error::Join(format!("{:?}", e)))?;
+
+    if !rx.load(Ordering::Acquire) {
+        Err(Error::Mount(
+            source.as_ref().to_path_buf(),
+            target,
+            nix::Error::EIO,
+        ))
+    } else {
+        Ok(())
+    }
+}
+
+/// When the size of mount options is bigger than one page, try to reduce the size by compressing
+/// the `lowerdir` option for overlayfs. The assumption is that lower directories for overlayfs
+/// often have a common prefix.
+fn compact_lowerdir_option(opts: &[String]) -> (Option<PathBuf>, Vec<String>) {
+    let mut n_opts = opts.to_vec();
+    // No need to compact if there is no overlay or only one lowerdir
+    let (idx, lower_opts) = match find_overlay_lowerdirs(opts) {
+        None => return (None, n_opts),
+        Some(v) => {
+            if v.1.len() <= 1 {
+                return (None, n_opts);
+            }
+            v
+        }
+    };
+
+    let idx = idx as usize;
+    let common_dir = get_longest_common_prefix(&lower_opts);
+    if common_dir.is_empty() || common_dir == "/" {
+        return (None, n_opts);
+    }
+
+    info!(
+        sl!(),
+        "compact_lowerdir_option get common prefix: {}", &common_dir
+    );
+    let lower: Vec<String> = lower_opts
+        .iter()
+        .map(|c| c.replace(&common_dir, ""))
+        .collect();
+    n_opts[idx as usize] = format!("lowerdir={}", lower.join(":"));
+
+    (Some(PathBuf::from(common_dir)), n_opts)
+}
+
+fn find_overlay_lowerdirs(opts: &[String]) -> Option<(usize, Vec<String>)> {
+    for (idx, o) in opts.iter().enumerate() {
+        if let Some(lower) = o.strip_prefix("lowerdir=") {
+            if !lower.is_empty() {
+                let c_opts: Vec<String> = lower.split(':').map(|c| c.to_string()).collect();
+                return Some((idx, c_opts));
+            }
+        }
+    }
+
+    None
+}
+
+fn get_longest_common_prefix(opts: &[String]) -> String {
+    if opts.is_empty() {
+        return String::new();
+    }
+
+    let mut paths = Vec::with_capacity(opts.len());
+    for opt in opts.iter() {
+        match Path::new(opt).parent() {
+            None => return "/".to_string(),
+            Some(v) => paths.push(v),
+        }
+    }
+
+    let mut res = match common_path::common_path_all(paths) {
```

Comment:
I'm not sure we should be using this crate: https://crates.io/crates/common_path shows the source as https://gitlab.com/pwoolcoc/common-path, which no longer exists, which suggests this may have been abandoned. The user is also on GH, but I don't see the crate there either: https://github.com/pwoolcoc.

Reply:
Reimplemented in another way, I'm also hesitating about the dependency when introducing it, it's not in a health state.